### ISSUE TITLE
Issue with Apple chip

### ISSUE THREAD

**Issue by jgarcia-sg** [microsoft/AzureTRE#2969]
Hi,

I tried to run this in my Apple Chip (arm64).

When doing

```
make build-resource-processor-vm-porter-image
```

It failed with the following error:

```
microsoft/AzureTRE-Deployment#9 5.326 Reading state information...
microsoft/AzureTRE-Deployment#9 5.409 E: Version '5:20.10.17~3-0~debian-buster' for 'docker-ce' was not found
microsoft/AzureTRE-Deployment#9 5.409 E: Unable to locate package docker-ce-cli
microsoft/AzureTRE-Deployment#9 5.409 E: Unable to locate package containerd.io
microsoft/AzureTRE-Deployment#9 5.409 E: Couldn't find any package by glob 'containerd.io'
microsoft/AzureTRE-Deployment#9 5.409 E: Couldn't find any package by regex 'containerd.io'
```

The issue was in this line in the vmss_porter dockerfile

```
&& echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/debian $(lsb_release -cs) stable" \
```

Since my architecture is arm64, it would not find the rpms needed to install and would fail with this error. I suggest to change it for:

```
 && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/debian $(lsb_release -cs) stable" \
```

**Comment by LizaShak** [comment]
@jgarcia-sg can you please confirm if you are building it within the dev container?

**Comment by jgarcia-sg** [comment]
Yes, I was building it inside the dev container.

**Comment by craddm** [comment]
Just wanted to add that I ran into this and several related issues when I first tried to deploy an Azure TRE using the dev container on an M1 Mac. The dockerfiles are typically agnostic as to the architecture used, with a few exceptions like this one where `amd64` is explicitly specified. Even though the dev container itself was running an `amd64` arch, subsequent images would be built as `arm64`, and then I'd need to modify the dockerfile to also install `gcc` etc. I partly solved it by explicitly making the dockerfiles use `amd64` platform architectures, but I kept running into problems while trying to actually deploy on Azure. In the end I just switched to deploying from an `amd64` VM instead.
